package.json

```diff
@@ -30,12 +30,20 @@
         "format": "prettier-eslint --write",
         "lint": "eslint --cache --max-warnings 0 --ext .ts,.js src",
         "precommit": "lint-staged",
-        "test": "jest --config=jest.json",
         "test:watch": "yarn test --watch",
-        "build": "npm-run-all --parallel build:declarations build:transpile",
+        "build": "npm-run-all --parallel build:**",
         "build:declarations": "tsc --p ./tsconfig.build.json",
-        "build:transpile": "babel src --ignore **/*.spec.ts --out-dir lib/cjs --extensions \".ts,.tsx\"",
+        "build:dist": "gulp build:dist build:dist:min",
+        "build:lib:cjs": "gulp build:lib:cjs",
+        "build:lib:esm": "gulp build:lib:esm",
+        "build:lib:umd": "gulp build:lib:umd build:lib:umd:min",
+        "prebuild:declarations": "rm -f lib/*.d.ts",
+        "prebuild:dist": "rm -rf dist",
+        "prebuild:lib:cjs": "rm -rf lib/cjs",
+        "prebuild:lib:esm": "rm -rf lib/esm",
+        "prebuild:lib:umd": "rm -rf lib/umd",
```

Comment:
I'm tempted to say that we should combine all the lib into one build:lib -- what would be the use case of wanting to build in one lib format but not the other? The time to build would be relatively incremental

Reply:
Yeah you're probably right. I'll think on it. I don't like the script explosion, but they're definitely gonna be separate in gulp. The kinda sucky part is that the declarations are also put in `lib/` so I can't just delete the whole `lib/` folder